### ISSUE TITLE
Friends

### ISSUE THREAD

**Issue by johansolbakken** [ThomasNitsche99/UltOrcs_v2#24]
Thinking of: adding a friends table to the database that connects a user with other users. Also adding api for becoming friends.

**Comment by johansolbakken** [comment]
@ThomasNitsche99 should it be like this: a user clicks invite to be friends, and then the other user can click accept?

**Comment by ThomasNitsche99** [comment]
Agree, the user should agree of becoming friends first. Any thoughts on how this can be done?

**Comment by johansolbakken** [comment]
Yes, I think I can make a table that links users. Ill will make branch now.

**Comment by johansolbakken** [comment]
In #25 I have implemented:
- posting a friend request
- getting a list of friend requests

Because I did not have access to user.friends, I could not implement:
- get a list of friends
- when accept, add users to each other's friend list

However, accept API will remove FriendRequest from the list.

**Comment by johansolbakken** [comment]
On this issue there is still these:
TODO: Actually accept friend requests and add them to the friend list
TODO: Remove the friend option maybe?
TODO: Get list of friends

**Comment by ThomasNitsche99** [comment]
Good work! Ill look into getting access to the types 🗡️ We should also include the option to remove friends!

**Comment by johansolbakken** [comment]
#30 added a table called Friendship. I have been able to make friendships in this commit, so I will use this going forward.
